JitsiConference.js

```diff
@@ -1623,7 +1786,519 @@ JitsiConference.prototype.broadcastEndpointMessage = function(payload) {
+ * @param {string} logName the session's nickname which will appear in log
+ * messages.
+ * @param {JingleSessionPC} jingleSession the session for which remote tracks
+ * will be added.
+ * @private
+ */
+JitsiConference.prototype._addRemoteTracks = function(logName, jingleSession) {
+    if (!jingleSession) {
+        logger.info(`Not adding remote ${logName} tracks - no session yet`);
+
+        return;
+    }
+    const remoteTracks = jingleSession.peerconnection.getRemoteTracks();
+
+    for (const track of remoteTracks) {
+        logger.info(`Adding remote ${logName} track: ${track}`);
+        this.rtc.eventEmitter.emit(RTCEvents.REMOTE_TRACK_ADDED, track);
+    }
+};
+
+/**
+ * Called when {@link XMPPEvents.CONNECTION_ESTABLISHED} event is
+ * triggered for a {@link JingleSessionPC}. Switches the conference to use
+ * the P2P connection if the event comes from the P2P session.
+ * @param {JingleSessionPC} jingleSession the session instance.
+ * @private
+ */
+JitsiConference.prototype._onP2PConnectionEstablished
+= function(jingleSession) {
+    if (!jingleSession.isP2P) {
+
+        return;
+    } else if (this.p2pJingleSession !== jingleSession) {
+        logger.error('CONNECTION_ESTABLISHED - not P2P session ?!');
+
+        return;
+    }
+
+    // Update P2P status and emit events
+    this._setP2PStatus(true);
+
+    // Remove remote tracks
+    this._removeRemoteJVBTracks();
+
+    // Add remote tracks
+    this._addRemoteP2PTracks();
+
+    // Remove local tracks from JVB PC
+    // But only if it has started
+    if (this.jvbJingleSession) {
+        this._detachLocalTracksFromJvbSession();
+    }
+
+    // Start remote stats
+    logger.info('Starting remote stats with p2p connection');
+    this._startRemoteStats();
+};
+
+/**
+ * Detaches local tracks from the JVB connection.
+ * @private
+ */
+JitsiConference.prototype._detachLocalTracksFromJvbSession = function() {
+    const localTracks = this.getLocalTracks();
+
+    logger.info(`Detaching local tracks from JVB: ${localTracks}`);
+    this.jvbJingleSession.detachLocalTracks(localTracks).then(
+        () => {
+            logger.info(`Detach local tracks from JVB done!${localTracks}`);
+        },
+        error => {
+            logger.info(
+                `Detach local tracks from JVB failed!${localTracks}`, error);
+        });
+};
+
+/**
+ * Clears the deferred start P2P task if it has been scheduled.
+ * @private
+ */
+JitsiConference.prototype._maybeClearDeferredStartP2P = function() {
+    if (this.deferredStartP2P) {
+        logger.info('Cleared deferred start P2P task');
+        clearTimeout(this.deferredStartP2P);
+        this.deferredStartP2P = null;
+    }
+};
+
+/**
+ * Removes from the conference remote tracks associated with the JVB
+ * connection.
+ * @private
+ */
+JitsiConference.prototype._removeRemoteJVBTracks = function() {
+    this._removeRemoteTracks('JVB', this.jvbJingleSession);
```

Comment:
Same as with the add methods. There is extra check and possible there could be more that would have to be duplicated otherwise.